### ISSUE TITLE
Point wise refinement (backreference)

### ISSUE THREAD

**Issue by JohanBergius** [xinge008/Cylinder3D#125]
Hi Xinge, Thanks for the work!  I was hoping to do point refinement by using hashmap to backreference to the original points. However I am not able to get the solution to converge this way! Any suggestions on how you manage to make it converge?

________________________CODE________________
        logits = self.logits(up0e) #torch.Size([1, 1, 480, 360, 32])
        y = logits.dense()
        cur_dev = y[0].get_device()
        
        lista = torch.ones(len(train_vox_ten[0])) #
        lista = lista[:,None]
        for i, q in enumerate(train_vox_ten):
            q = q[0]
            lista[i,:] = y[0,:,q[0].item(),q[1].item(),q[2].item()]

**Comment by xinge008** [comment]
you can check this issue https://github.com/xinge008/Cylinder3D/issues/28
